### PR TITLE
feat: download csv

### PR DESCRIPTION
**Pull request by lucafurrer**
Allow the download of uploaded CSVs

### CODE REVIEW

File: apis/core/lib/handlers/csv-source.ts [hunk 1]
```diff
@@ -75,7 +75,8 @@ const getCSVSource: express.RequestHandler = asyncMiddleware(async (req, res, ne
     return next(new Error('s3 key not found'))
   }
 
-  res.redirect(directDownload)
+  res.header('Location', directDownload)
```
Comment (lucafurrer) [inline apis/core/lib/handlers/csv-source.ts:78]: @tpluscode I run into CORS-Problems using redirect and still wondering if there is a better way than this.

File: ui/src/api/mixins/CSVSource.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ export default function mixin<Base extends Constructor> (base: Base): Mixin {
     get _additionalActions () {
       return {
         replace: ns.cc.ReplaceCSVAction,
+        download: ns.cc.GetCSVSourceAction
```
Comment (tpluscode) [inline ui/src/api/mixins/CSVSource.ts:18]:
```suggestion
        download: schema.DownloadAction
```

File: packages/core/namespace.ts [hunk 3]
```diff
@@ -28,7 +28,8 @@ type CubeCreatorClass =
   'Observations' |
   'TransformJob' |
   'PublishJob' |
-  'ReplaceCSVAction'
+  'ReplaceCSVAction' |
+  'GetCSVSourceAction'
```
Comment (tpluscode) [inline packages/core/namespace.ts:32]: Won't need a custom type. Can use `schema` (see other comment)

File: ui/src/components/CsvSourceMapping.vue [hunk 4]
```diff
@@ -186,6 +191,11 @@ export default class CsvSourceMapping extends Vue {
     })
   }
 
+  async downloadSource (source: CsvSource): Promise<void> {
+    const downloadLink = await this.$store.dispatch('project/downloadCSV', source)
+    window.open(downloadLink)
```
Comment (tpluscode) [inline ui/src/components/CsvSourceMapping.vue:196]: I don't think this belongs to the component. Please move `window.open` to `store/modules/project.ts`

File: apis/core/hydra/csv-source.ttl [hunk 5]
```diff
@@ -43,7 +43,7 @@ cc:CSVSource
         hydra:Operation,
         cc:GetCSVSourceAction ;
       hydra:title
-        "Get CSV Source" ;
+        "Get CSV" ;
```
Comment (tpluscode) [inline apis/core/hydra/csv-source.ttl:46]:
```suggestion
        "Download CSV" ;
```

maybe?

**Review by tpluscode — CHANGES_REQUESTED** [review]

**Pull request merged**